media-gfx/gscan2pdf/gscan2pdf-2.5.3.ebuild

```diff
@@ -0,0 +1,59 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit eutils perl-module
+
+DESCRIPTION="Scan documents, perform OCR, produce PDFs and DjVus"
+HOMEPAGE="http://gscan2pdf.sourceforge.net/"
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.xz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+# OCR tests fail with tesseract[opencl], not fixed by addpredict
```

Comment:
That doesn't sound like a very common configuration. Why not just test-DEPEND on `tesseract[-opengl]`? Or use `has_version` to conditionally skip tests in `src_test`.

Reply:
Yes it is a bit of an excuse. It used to be that tests were frequently breaking on new builds so it wasn't realistic to ship, especially if you follow the convention of not shipping patches that are not upstreamed, upsteam repo tends to get updated at the time of the next release.

Tests seem to be more stable now but ImageMagick restrictions have caused more complication. So this is what it takes to run the tests:

```diff
@@ -3,7 +3,9 @@
 
 EAPI=6
 
-inherit eutils perl-module
+DIST_TEST="do"
+
+inherit eutils perl-module virtualx
 
 DESCRIPTION="Scan documents, perform OCR, produce PDFs and DjVus"
 HOMEPAGE="http://gscan2pdf.sourceforge.net/"
@@ -13,8 +15,7 @@
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
 
-# OCR tests fail with tesseract[opencl], not fixed by addpredict
-RESTRICT="test"
+IUSE="test"
 
 RDEPEND="
 	dev-perl/Config-General
@@ -46,6 +47,22 @@
 	media-gfx/sane-backends
 	media-libs/tiff"
 
+DEPEND="test? (
+	$RDEPEND
+	dev-perl/Sub-Override
+	media-libs/fontconfig
+
+	app-text/djvu[tiff]
+	app-text/poppler[utils]
+	app-text/tesseract[-opencl,osd,tiff]
+	app-text/unpaper
+	media-gfx/imagemagick[djvu,png,tiff,perl]
+	media-gfx/sane-backends[sane_backends_test]
+	media-gfx/sane-frontends
+	)"
+
+PERL_RM_FILES=( t/{90_MANIFEST,91_critic,99_pod}.t )
+
 mydoc="History"
 
 pkg_postinst() {
@@ -57,3 +74,16 @@
 	optfeature "automatic document feeder support" media-gfx/sane-frontends
 	optfeature "sending PDFs as email attachments" x11-misc/xdg-utils
 }
+
+src_test(){
+	local confdir="${HOME}/.config/ImageMagick"
+	mkdir -p ${confdir} || die
+	cat > ${confdir}/policy.xml <<-EOT
+		<policymap>
+		<policy domain="coder" rights="read|write" pattern="PDF" />
+		<policy domain="coder" rights="read" pattern="PS" />
+		</policymap>
+	EOT
+	eend 0
+	virtx perl-module_src_test
+}
```

Reply:
1. `cat > ${confdir}/policy.xml <<-EOT || die`
2. eend is meaningless
3. `${RDEPEND}`

Other than that, looks fine.

Reply:
All 3 sorted.